### PR TITLE
Update ocaml to version 4.11.1

### PR DESCRIPTION
**Pull request by ImperatorS79**

### CODE REVIEW

File: mingw-w64-ocaml/PKGBUILD [hunk 1]
```diff
@@ -2,73 +2,124 @@
...
+  cd "${srcdir}/${_realname}-${pkgver}"
+  patch -p1 -i "${srcdir}/0001-fix-misc-h.patch"
+}
+
+build() {  
+  cd "${srcdir}/${_realname}-${pkgver}"
+  
+  #flexlink needs ocaml, ocaml needs flexlink
+  rm -rf flexdll/
+  wget https://github.com/alainfrisch/flexdll/archive/0.39.tar.gz
```
Comment (lazka) [inline mingw-w64-ocaml/PKGBUILD:36]: Add it to `source` instead of manually downloading it.

File: mingw-w64-ocaml/PKGBUILD [hunk 2]
```diff
@@ -2,73 +2,124 @@
...
+makedepends=('wget' 'tar'
+             "${MINGW_PACKAGE_PREFIX}-ncurses"
+             "${MINGW_PACKAGE_PREFIX}-gcc"
+             "${MINGW_PACKAGE_PREFIX}-crt-git")
+optdepends=("${MINGW_PACKAGE_PREFIX}-ncurses: advanced ncurses features"
+            "${MINGW_PACKAGE_PREFIX}-tk: advanced tk features")
+source=("https://caml.inria.fr/distrib/ocaml-${pkgver%.*}/${_realname}-${pkgver}.tar.xz"
+        "0001-fix-misc-h.patch")
+sha256sums=('ea4a9187035019f219a50458b7c09ab74239272e5b6ea3ab2a0c840b5810854c'
+            'SKIP')
```
Comment (lazka) [inline mingw-w64-ocaml/PKGBUILD:22]: don't skip

File: mingw-w64-ocaml/PKGBUILD [hunk 3]
```diff
@@ -2,73 +2,124 @@
...
 pkgrel=1
 pkgdesc="An industrial strength programming language supporting functional, imperative and object-oriented styles (mingw-w64)"
 arch=('any')
 url='https://ocaml.org/'
 license=('Q Public Licence 1.0' 'GPL2')
-depends=("${MINGW_PACKAGE_PREFIX}-flexdll")
-makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
-             "diffutils")
-source=("https://caml.inria.fr/pub/distrib/ocaml-${pkgver%.*}/${_realname}-${pkgver}.tar.gz"
-        "0002-Adapt-to-msys2-directory-layout.patch"
```
Comment (lazka) [inline mingw-w64-ocaml/PKGBUILD:16]: I guess all those patch files need to be removed as well?

**Comment by lazka** [comment]
Thanks. Just wondering, why do we need two packages here?

**Comment by ImperatorS79** [comment]
Since archlinux does it that way I did it that way too.

**Pull request merged**